src/error.rs

```diff
@@ -62,6 +73,112 @@ impl Error {
         self.status = status;
     }
 
+    /// Get a mutable reference to a header.
+    pub fn header_mut(&mut self, name: impl Into<HeaderName>) -> Option<&mut HeaderValues> {
+        self.headers.as_mut()?.get_mut(name.into())
+    }
+
+    /// Get an HTTP header.
+    pub fn header(&self, name: impl Into<HeaderName>) -> Option<&HeaderValues> {
+        self.headers.as_ref()?.get(name.into())
+    }
+
+    /// Remove a header.
+    pub fn remove_header(&mut self, name: impl Into<HeaderName>) -> Option<HeaderValues> {
+        self.headers.as_mut()?.remove(name.into())
+    }
+
+    /// Set an HTTP header.
+    pub fn insert_header(
+        &mut self,
+        name: impl Into<HeaderName>,
+        values: impl ToHeaderValues,
+    ) -> Option<HeaderValues> {
+        if self.headers.is_none() {
+            self.headers = Some(Headers::new());
+        }
+        self.headers.as_mut()?.insert(name, values)
+    }
+
+    /// Append a header to the headers.
+    pub fn append_header(&mut self, name: impl Into<HeaderName>, values: impl ToHeaderValues) {
+        if self.headers.is_none() {
+            self.headers = Some(Headers::new());
+        }
+        if let Some(headers) = &mut self.headers {
+            headers.append(name, values);
+        }
+    }
+
+    /// Take the headers.
+    pub fn take_headers(&mut self) -> Option<Headers> {
+        mem::replace(&mut self.headers, None)
```

Comment:
This doesn't seem right but I don't know how to properly free the ownership otherwise...

Reply:
Can you use Option::take?